app/charts/index.ts

```diff
@@ -639,37 +639,96 @@ const chartConfigsAdjusters: ChartConfigsAdjusters = {
 };
 type ChartConfigAdjusters = typeof chartConfigsAdjusters[ChartType];
 
-// Needed to correctly retain chart options when switching to maps (and tables?).
+// Needed to correctly retain chart options when switching to maps and tables.
 const chartConfigsPathOverrides: {
-  [chartType in ChartType]: {
-    [fieldToOverride: string]: string;
+  [newChartType in ChartType]: {
+    [oldChartType in ChartType]?: {
+      [oldFieldToOverride: string]: string;
+    };
   };
 } = {
   bar: {},
   column: {
+    map: {
     "fields.areaLayer.componentIri": "fields.x.componentIri",
     "fields.areaLayer.measureIri": "fields.y.componentIri",
+    },
+    table: {
+      fields: "fields.segment",
+    },
   },
   line: {
+    map: {
     "fields.areaLayer.measureIri": "fields.y.componentIri",
+    },
+    table: {
+      fields: "fields.segment",
+    },
   },
   area: {
+    map: {
     "fields.areaLayer.measureIri": "fields.y.componentIri",
+    },
+    table: {
+      fields: "fields.segment",
+    },
   },
   scatterplot: {
+    map: {
     "fields.areaLayer.measureIri": "fields.y.componentIri",
+    },
+    table: {
+      fields: "fields.segment",
+    },
   },
   pie: {
+    map: {
     "fields.areaLayer.componentIri": "fields.x.componentIri",
     "fields.areaLayer.measureIri": "fields.y.componentIri",
+    },
+    table: {
+      fields: "fields.segment",
+    },
+  },
+  table: {
+    column: {
+      "fields.segment": "fields",
+    },
+    line: {
+      "fields.segment": "fields",
+    },
+    area: {
+      "fields.segment": "fields",
+    },
+    scatterplot: {
+      "fields.segment": "fields",
+    },
+    pie: {
+      "fields.segment": "fields",
+    },
```

Comment:
Nit: wondering if there should be a `default` destination chart so that we can remove some duplication here.

Reply:
Hmmm, I am not sure about that, as we exclude Map chart in this particular case, so we still would need to define either exclusions (Map) or inclusions (all expect Map). In fact, no override field is defined for all chart types, so not sure is there should be a default one to fallback to, as in such cases it should return undefined to proceed normally. What do you think?

Reply:
OK I agree, thanks.